### PR TITLE
command line interface app

### PR DESCRIPTION
**Pull request by ghifo**
The remove function does not work as supposed yet, I am still working on it.

### CODE REVIEW

File: week2/homework/src/index.js [hunk 1]
```diff
@@ -1,3 +1,161 @@
...
+      } else {
+        console.log(error);
+      }
+    } else {
+      if (num < 0 || num > todoList.length) {
+        console.log('Please write a valid number');
+      } else {
+        const splitList = todoList.split('\n');
+        // make the list an array of lines
+        const newList = splitList.filter((elem, index) => index !== num - 1).join('\n');
```
Comment (eidosam) [inline week2/homework/src/index.js]: May [splice](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/splice) serves this case better than `filter`?

File: week2/homework/src/index.js [hunk 2]
```diff
@@ -1,3 +1,161 @@
...
+        console.log(newList);
+        fs.writeFile('./src/todoList.txt', newList, error => console.error(error));
```
Comment (eidosam) [inline week2/homework/src/index.js]: Print the error message only when there is. Same way you did in `add` function, lines (57-61

File: week2/homework/src/index.js [hunk 3]
```diff
@@ -1,3 +1,161 @@
...
+      } else {
+        console.log(error);
+      }
+    } else {
+      if (num < 0 || num > todoList.length) {
+        console.log('Please write a valid number');
+      } else {
+        const splitList = todoList.split('\n');
+        console.log(num);
+        console.log(newVal);
```
Comment (eidosam) [inline week2/homework/src/index.js]: Do you need to show the user these messages?

File: week2/homework/src/index.js [hunk 4]
```diff
@@ -1,3 +1,161 @@
...
+
+        // make the list an array of lines
+        const newList = splitList
+          .map((elem, index) => (index === num - 1 ? newVal : elem))
+          .join('\n');
+        // console.log(newList);
+        fs.writeFile('./src/todoList.txt', newList, error => console.error(error));
```
Comment (eidosam) [inline week2/homework/src/index.js]: Same note about handling/printing the errors.

File: week2/homework/src/index.js [hunk 5]
```diff
@@ -1,3 +1,161 @@
...
+      }
+    }
+  });
+}
+function reset() {
+  fs.writeFile('./src/todoList.txt', '', (error, todoList) => {
+    if (error) {
+      if (error.code === 'ENOENT') {
+        console.log('no data found');
```
Comment (eidosam) [inline week2/homework/src/index.js]: Error `ENOENT` occurs in `fs.writeFile` when the parent directory of the file is not present. So this message is invalid.

File: week2/homework/src/index.js [hunk 6]
```diff
@@ -1,3 +1,161 @@
...
+        console.log(error);
+      }
+    } else {
+      console.log(`Rest!
+      Nothing to be done in the list`);
+    }
+  });
+}
+function help() {
+  fs.readFile('./src/todoList.txt', 'utf8', error => {
```
Comment (eidosam) [inline week2/homework/src/index.js]: Help command is only for printing the usage guide, nothing else. So no need to check the existence of the data file.

File: week2/homework/src/index.js (hunk 5 above)
Comment (ghifo) [reply week2/homework/src/index.js]: Shall I write the error this way, or no need at all for using error hier?
 if (error) {
      console.log(error);
    }

File: week2/homework/src/index.js (hunk 1 above)
Comment (ghifo) [reply week2/homework/src/index.js]: Is there any problem of using filter here?
I was advised to use it :)

File: week2/homework/src/index.js (hunk 5 above)
Comment (eidosam) [reply week2/homework/src/index.js]: That should be good enough. As you suggested

    if (error) {
        console.log(error);
    } else {
        console.log(`Rest!
            Nothing to be done in the list`);
    }

File: week2/homework/src/index.js (hunk 1 above)
Comment (eidosam) [reply week2/homework/src/index.js]: Not a problem, but it is different use case. Usually `filter` is used to filter data on the content of the array, not the index of the elements.

File: week2/homework/src/index.js (hunk 5 above)
Comment (ghifo) [reply week2/homework/src/index.js]: Thanks

**Pull request closed without merging**